hackathon/templates/hackathon/hackathon_detail.html

```diff
@@ -0,0 +1,86 @@
+{% extends "base.html" %}
+{% load static %}
+{% block content %}
+<section class="h-100">
+    <h2 class="p-orange text-center">{{ hackathon.display_name }}</h2>
+    <article class="card shadow mb-3">
+        <div class="card-body">
+            <p class="card-text">{{ hackathon.description }}</p>
+            <h6 class="card-subtitle mb-2 text-muted"><strong>From</strong> {{ hackathon.start_date }} <strong>to</strong> {{ hackathon.end_date }}</h6>
+            {% if hackathon.organiser %}
+                <p>Organiser: <a href="#" class="p-blue card-link">{{ hackathon.organiser }}</a></p>
+            {% endif %}
+            {% if request.user.is_staff %}
+                {% include 'hackathon/includes/enroll.html' %}
+            {% endif %}
+            <a href="{% url 'hackathon:hackathon-list' %}" class="btn btn-ci">See other Hackathons</a>
+        </div>
+    </article>
+</section>
+{% endblock %}
+{% block js %}
+{{ block.super }}
+{% if request.user.is_staff %}
+    <script>
```

Comment:
Could you extract the JS content to an external file please?

Reply:
Also, why have you gone with an AJAX approach here instead of just submitting the form and redirecting back to the updated page? Since we are using Django I find this a bit counter intuitive.

Reply:
Okay, cool. I'll outline my AJAX reasoning below and you make the call.

Will I move it to the main `script.js` or will I create a different one? I wasn't sure where it would be more appropriate so I used the `block js`.

I went with AJAX because it's easy and fast. Some buttons do pretty much the same thing, only adding/removing info from the database, so it's simpler. In other user stories, like "enroll/withdraw as a participant", it would be easy to add another button on the page that would function similarly, using the same scripts. Some users would have two buttons. This way it's more like clickable options in front of you, instead of sending numerous forms. I think it's nice.

From a user perspective, it's a smoother, more responsive experience. The less required page loading, the better, no? A page-load makes sense for detailed forms or new content, but it's a small update and doesn't provide the user with any new content or info. So, I thought "why reload the whole page?". I got a lot of feedback on my final milestone about users not wanting to stay on sites with page-reloads that don't provide new content or engagement and I ended up using a lot of ajax. In summary, I thought it was a useful addition for UX.

I'm wondering why is it counter-intuitive with Django? 
If you think it's better without it, I'll take it out, still, it would be useful for me to learn.

Also, I've realised I need to add code to prevent users from signing up to Hackathons past their end date, so I have to do that too before I've finished.

Reply:
@Ri-Dearg okay fair enough. It is a bit counter intuitive for me, but I can see your point and thinke we can keep the AJAX logic then. In terms of the external JS. I'd just add it to `script.js` since we don't really have any JS at the moment it should be fine.